src/fhirtypes/ElementDefinition.ts

```diff
@@ -1642,8 +1643,14 @@ export class ElementDefinition {
    * @throws {InvalidUriError} when the system being assigned is not a valid uri
    */
   private assignFshCode(code: FshCode, exactly = false, fisher?: Fishable): void {
-    if (code.system && !isUri(code.system.split('|')[0])) {
-      throw new InvalidUriError(code.system);
+    if (code.system) {
+      const csURI = code.system.split('|')[0];
+      const vsURI = fisher?.fishForMetadata(code.system, Type.ValueSet)?.url ?? '';
+      if (vsURI) {
+        throw new MismatchedBindingTypeError(code.system, this.path, 'Code System');
```

Comment:
I notice that here you use "Code System", but in the other spot you use "ValueSet". Would you be able to update this to be "CodeSystem"? I think I'd live a happier life if I could just let that " " go, but oh well.